File: leetcode/100-150/119-isPowerOfThree.java
    public boolean isPowerOfThree(int n) {
        while (n>0 && n%3==0) n/=3;
        return n==1;
    }

    public boolean isPowerOfThree2(int n) {
        return n>0 && (n==1 || (n%3==0 && isPowerOfThree(n/3)));
    }

    public boolean isPowerOfThree3(int n) {
        return Integer.toString(n, 3).matches("^10*$");
    }

File: leetcode/100-150/119-isPowerOfThree.py
def isPowerOfThree(self, n):
    while n>0 and n%3==0: n/=3
    return n==1

def isPowerOfThree2(self, n):
    return n>0 and (n==1 or (n%3==0 and self.isPowerOfThree(n/3)))

def isPowerOfThree3(self, n):
    return n>0 and 1162261467%n==0